### DISCUSSION TITLE
Client ip

### DISCUSSION THREAD

**Discussion by mathysth** [honojs/hono#2558] [General]
How can I get the client ip, I didn't found the after looking the api

**Comment by yasuaki640** [comment]
If you are using Bun, then you can use the method described in [the documentation to get the IP address.](https://hono.dev/getting-started/bun#obtaining-request-ip-address)

However, (as far as I can tell from the documentation) there is no generic API for getting an IP address, and it seems that the only way to get an IP address at this point is to use a method that is appropriate for the environment you are running in.

For example, Cloudflare workers can get the client IP in [this document](https://developers.cloudflare.com/fundamentals/reference/http-request-headers/#cf-connecting-ip)
